sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -641,12 +641,16 @@ class GpuOrcPartitionReader(
       }
       isFirstBatch = false
     }
-    batch.isDefined
+    val ret = batch.isDefined
+    // After finishing reading, we should clean up ctx just in case leaking orc readers
+    if (!ret) {
+      cleanUpOrc(ctx)
+    }
+    ret
   }
 
   override def close(): Unit = {
     super.close()
-    cleanUpOrc(ctx)
```

Comment:
Why wouldn't we clean up the reader when this is closed?  If the code downstream from the iterator ends up stopping early and not draining isn't drained and we never see the empty batch from the reader -- how does the orc reader get closed in that case?

Reply:
Thx Jason for the findings. You are correct. Done.